### PR TITLE
duckdb: add v0.9.2

### PR DESCRIPTION
**Pull request by glentner**
Update to add latest release. Also add myself as maintainer.

### CODE REVIEW

**Review by tldahlgren — DISMISSED** [review]
Confirmed the new version sha256.

**Comment by tldahlgren** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/duckdb/package.py [hunk 1]
```diff
@@ -11,11 +11,17 @@ class Duckdb(CMakePackage):
...
     git = "https://github.com/duckdb/duckdb.git"
 
     license("MIT")
+    maintainers("glentner")
 
     version("master", branch="master")
+    version(
+        "0.9.2",
+        sha256="afff7bd925a98dc2af4039b8ab2159b0705cbf5e0ee05d97f7bb8dce5f880dc2",
+        preferred=True,
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/duckdb/package.py]: No need to add preferred, if the version is the latest according to our sort algorithm.

Comment (teaguesterling) [reply var/spack/repos/builtin/packages/duckdb/package.py]: I've also made a pull request to add variants for duckdb extensions and address a version identification issue in the installer here: https://github.com/spack/spack/pull/38922. Happy to share the maintainer responsibility with you @glentner.

File: var/spack/repos/builtin/packages/duckdb/package.py [hunk 2]
```diff
@@ -11,11 +11,17 @@ class Duckdb(CMakePackage):
...
+    )
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/duckdb/package.py]:
```suggestion
    version("0.9.2", sha256="afff7bd925a98dc2af4039b8ab2159b0705cbf5e0ee05d97f7bb8dce5f880dc2")
```

**Review by alalazo — APPROVED** [review]

**Pull request merged**